client/src/pages/Settings/Availability/Availability.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState, useEffect } from 'react';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Paper from '@material-ui/core/Paper';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import useStyles from './useStyles';
+import Typography from '@material-ui/core/Typography';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { useAuth } from '../../../context/useAuthContext';
+import Button from '@material-ui/core/Button';
+import Drawer from '@material-ui/core/Drawer';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import DateForm from './DateForm/DateForm';
+import { Divider } from '@material-ui/core';
+import getAvailability from '../../../helpers/APICalls/getAvailability';
+import updateAvailability from '../../../helpers/APICalls/updateAvailability';
+import { Availability, date } from '../../../interface/Availability';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+
+export default function Availiability(): JSX.Element {
+  const [loading, setLoading] = useState(false);
+  const { updateSnackBarMessage } = useSnackBar();
+  const [availability, setAvailability] = React.useState<Availability>({
+    Monday: {
+      to: '',
+      from: '',
+    },
+    Tuesday: {
+      to: '',
+      from: '',
+    },
+    Wednesday: {
+      to: '',
+      from: '',
+    },
+    Thursday: {
+      to: '',
+      from: '',
+    },
+    Friday: {
+      to: '',
+      from: '',
+    },
+    Saturday: {
+      to: '',
+      from: '',
+    },
+    Sunday: {
+      to: '',
+      from: '',
+    },
+  });
```

Comment:
This can be reduced all down to `React.useState<Availability>({} as Availability);`

Reply:
OK

Reply:
We should be utilizing Formik here and not relying on making state like this for form data.

Reply:
Nevermind your original solution worked. I was not accurately checking if the useEffect was completed before rendering the form. I've used your original solution